python/ql/src/semmle/python/objects/TObject.qll

```diff
@@ -367,6 +367,8 @@ predicate receiver(AttrNode instantiation, PointsToContext context, ObjectIntern
 pragma [noinline]
 private predicate self_parameter(ParameterDefinition def, PointsToContext context, PythonClassObjectInternal cls) {
     def.isSelf() and
+    /* Exclude the special parameter name `.0` which is used for unfolded comprehensions. */
+    def.getName() != ".0" and
     exists(Function scope |
         def.getScope() = scope and
         def.isSelf() and
```

Comment:
Don't you think that two checks to `def.isSelf` is overdoing it? :smile:

Reply:
Indeed. I've remove the second instance.